### PR TITLE
[YVR-Server-65] Extract zone code from csv file

### PR DESCRIPTION
**Pull request by ys421**
https://trello.com/c/k2hv634o

### CODE REVIEW

File: app/components/FileUpload.jsx [hunk 1]
```diff
@@ -23,9 +24,10 @@ const FileUpload = () => {
           console.log(err);
         } else {
           console.log(data);
+          //Zone Code List : Additional list of zone codes will be added manually
           console.log(calculateHourlyAverages(data))
 
-          //Zone Code List : Additional list of zone codes will be added manually
+          //Extracting Zone Code from the data
           const zoneCodeList = [
```
Comment (gdcho) [inline app/components/FileUpload.jsx:33]: @jasper-oh Do we have the entire list of all the preassigned zones for zoneCodeList?

**Comment by jasper-oh** [comment]
Checking the code

File: app/components/FileUpload.jsx [hunk 2]
```diff
@@ -9,27 +9,73 @@ const fileTypes = ["csv"]
...
-          console.log(data);
-          //TODO YS's algorithm will be implemented here
-          console.log(calculateHourlyAverages(data))
+          //Zone Code List : Additional list of zone codes will be added manually
+
+          const dataToStore = calculateHourlyAverages(data)
+          setDataToUpload(dataToStore);
 
+          //Extracting Zone Code from the data
+          const zoneCodeList = [
```
Comment (jasper-oh) [inline app/components/FileUpload.jsx:33]: Will update this logic later.

Still having a doubt about the const list in here.

Comment (gdcho) [reply app/components/FileUpload.jsx:33]: Seems hard coded - will not work if zone names are changed in the future

**Comment by jasper-oh** [comment]
- Didn't handle the exception when trying to upload the same csv file in this page.
1. Will create the ticket that handle various exceptions.

Besides that, looks good to me.

2. Will create the ticket that implement the function in Client code that fetch the data with endpoint with GET method
: api/data/{targetDate}/{targetAreaCode}
: api/data/{targetAreaCode}


the basic format of json response :

{
  "_id" : "ITB_VAV_A42_W3327_02_2024-03-24_01",
  "area_id" : "ITB_VAV_A42_W3327_02",
  "timestamp" : 2024-03-24T01:00:00.000+00:00 (DateTime),
  "temperature" : 21.56865505 (float),
  "air_flow" : 92.711746225 (float),
  "co2" : 556.76658625 (float)
}

@gdcho Can you check the logic one more time?

**Review by jasper-oh — APPROVED** [review]

**Comment by gdcho** [comment]
> 2\. Will create the ticket that implement the function in Client code that fetch the data with endpoint with GET method

Logic looks good

**Review by gdcho — APPROVED** [review]

File: app/components/FileUpload.jsx (hunk 1 above)
Comment (jasper-oh) [reply app/components/FileUpload.jsx:33]: Will create the ticket about this issue and assigned to me. will fix by today

**Pull request merged**